ref_builder/schema.py

```diff
@@ -29,3 +42,40 @@ class OTUSchema(BaseModel):
     @computed_field
     def multipartite(self) -> bool:
         return len(self.segments) > 1
+
+
+def set_segment_prefix(moltype: NCBISourceMolType):
+    if moltype in (
+        NCBISourceMolType.GENOMIC_DNA,
+        NCBISourceMolType.OTHER_DNA,
+        NCBISourceMolType.UNASSIGNED_DNA,
+    ):
+        return "DNA"
+
+    match moltype:
+        case NCBISourceMolType.GENOMIC_RNA:
+            return "RNA"
+        case NCBISourceMolType.MRNA:
+            return "mRNA"
+        case NCBISourceMolType.TRNA:
+            return "tRNA"
+        case NCBISourceMolType.TRANSCRIBED_RNA:
+            return "RNA"
+        case NCBISourceMolType.VIRAL_CRNA:
+            return "cRNA"
+        case NCBISourceMolType.OTHER_RNA:
+            return "RNA"
+
+
+complex_name_pattern = re.compile(r"([A-Za-z]+)[-_ ]+([A-Za-z0-9]+)")
+
+
+def parse_segment_name(raw: str) -> SegmentName:
```

Comment:
Could this just be merged into `get_multipartite_segment_name`?

Reply:
I can move it under `otu.utils` for now. I would like to be able to test it without `NCBIGenbank` for the time being.